tests/test_packages/test_skills/test_carpark.py

```diff
@@ -19,151 +19,49 @@
 
 """This test module contains the integration test for the weather skills."""
 
-import io
 import os
-import shutil
 import signal
-import subprocess  # nosec
-import sys
-import tempfile
-import threading
 import time
 
 import pytest
 
-from aea.cli import cli
-from aea.test_tools.click_testing import CliRunner
+from aea.test_tools.test_cases import AEAWithOefTestCase
 
-from ...conftest import AUTHOR, CLI_LOG_OPTION
 
-
-def _read_tty(pid: subprocess.Popen):
-    for line in io.TextIOWrapper(pid.stdout, encoding="utf-8"):
-        print("stdout: " + line.replace("\n", ""))
-
-
-def _read_error(pid: subprocess.Popen):
-    for line in io.TextIOWrapper(pid.stderr, encoding="utf-8"):
-        print("stderr: " + line.replace("\n", ""))
-
-
-class TestCarPark:
+class TestCarPark(AEAWithOefTestCase):
     """Test that carpark skills work."""
 
-    @pytest.fixture(autouse=True)
-    def _start_oef_node(self, network_node):
-        """Start an oef node."""
-
-    @classmethod
-    def setup_class(cls):
-        """Set up the test class."""
-        cls.runner = CliRunner()
-        cls.agent_name_one = "my_carpark_aea"
-        cls.agent_name_two = "my_carpark_client_aea"
-        cls.cwd = os.getcwd()
-        cls.t = tempfile.mkdtemp()
-        os.chdir(cls.t)
-
     def test_carpark(self, pytestconfig):
         """Run the weather skills sequence."""
         if pytestconfig.getoption("ci"):
```

Comment:
This should be moved to the parent class Oeftestcase

Reply:
Any idea how can we move it (in parent class)?
IMO good way to optimize this - to write a decorator and use it on every test we need to skip in CI.